### DISCUSSION TITLE
How to store tsl if current timestamp is less than last save timestamp due to RTC loss on embedded device

### DISCUSSION THREAD

**Discussion by amuroh** [armink/FlashDB#224] [Q&A] [answered]
Hello and congratulations for this great project.

I am integrating FlashDB TSL on a battery powered device. Sometimes on system crash or power loss the RTC may be reset to a default value (e.g. January 1st 2010). If I call tsl_append() it returns an error because current time is less than last saved timestamp.

`[FlashDB][tsl][log][fdb_tsdb1] Warning: current timestamp (ld) is less than or equal to the last save timestamp (ld). This tsl will be dropped.`

Is there any way to allow FlashDB to store TSL swith the current RTC time even if it is less than the last one?

**Comment by armink** [comment, marked as answer]
Hi, you can reimplement the get_time function and use count ++ instead of timestamp. After restart, the old count can be get by FDB_TSDB_CTRL_GET_LAST_TIME command .

**Reply by amuroh** [reply]
Thanks!
